lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb

```diff
@@ -265,16 +265,21 @@ fc_build_auxiliary_coordinate_latitude
         facts_cf.auxiliary_coordinate($coordinate)
         check is_latitude(engine, $coordinate)
         check not is_rotated_latitude(engine, $coordinate)
+        facts_cf.grid_mapping($grid_mapping)
+        check is_grid_mapping(engine, $grid_mapping, CF_GRID_MAPPING_LAT_LON)
     assert
         python cf_coord_var = engine.cf_var.cf_group.auxiliary_coordinates[$coordinate]
+        python cf_grid_var = engine.cf_var.cf_group.grid_mappings[$grid_mapping]
+        python coordinate_system = build_coordinate_system(cf_grid_var)
         python build_auxiliary_coordinate(engine, cf_coord_var,
-                                          coord_name=CF_VALUE_STD_NAME_LAT)
+                                          coord_name=CF_VALUE_STD_NAME_LAT,
+                                          coord_system=coordinate_system)
```

Comment:
I'm not absolutely clear how Pyke works, but I'm wondering if the `foreach` will loop over every `$coordinate`/`$gridmapping` pair, creating duplicate coordinates with grid mappings with which they are not associated. I think there needs to be an extra check, something like `check is_coordinate_grid_mapping(engine, $coordinate, $grid_mapping)` which returns true if `$grid_mapping` is the correct grid mapping for `$coordinate`